### PR TITLE
Ace fit params with yaml

### PR DESCRIPTION
**Pull request by bernstei**
save ace fitting parameters for `ace_fit.jl` as a yaml, not a json, so integer dict keys will not be converted to strings.

### CODE REVIEW

File: wfl/fit/ace.py [hunk 1]
```diff
@@ -132,14 +129,17 @@ def prepare_params(ACE_name, fitting_configs, ace_fit_params, run_dir='.', ref_p
...
 
-    if weight_from_sigma:
-        ace_fit_params["weights"] = {}
-        for at_i, at in enumerate(fitting_configs):
-            config_type = f"config_{at_i}"
-            at.info["config_type"] = config_type
-            ace_fit_params["weights"][config_type] = { "E": 1.0 / at.info.get("energy_sigma", 1.0),
-                                                       "F": 1.0 / at.info.get("force_sigma", 1.0),
-                                                       "V": 1.0 / at.info.get("virial_sigma", 1.0) }
+    from_sigma = ace_fit_params.get("weights", {}).get("from_sigma")
```
Comment (gelzinyte) [inline wfl/fit/ace.py:132]: Is this documented somewhere? I think it would be good to briefly mention?

Comment (gelzinyte) [reply wfl/fit/ace.py:132]: Maybe just in a docstring of this function?

**Comment by gelzinyte** [comment]
Looks good I think. The only other thing to watch out for is the params entries that in Julia are of tuple-to-tuple type. But those are a bit rarer and we don't modify them in the workflow fitting script, so I think we can leave it for users to take care of. 

Just to mention, if these tuples are saved (from python or other to yaml) as strings (e.g. `{"(C, C)" : "(1.2, 4)"}`), these get automatically parsed appropriately ( to `Dict((C, C) => (1.2, 4))`) in `ACE1pack.fill_defaults()`, at least when read from yaml.

File: wfl/fit/ace.py (hunk 1 above)
Comment (gelzinyte) [reply wfl/fit/ace.py:132]: Never mind - left a comment in the original PR.

**Comment by bernstei** [comment]
@gelzinyte what do you think about merging this one, since it's clearly safer than using JSON, regardless of whether ACE actually gets automatic conversion of integer-as-string keys?

**Comment by bernstei** [comment]
I guess since JSON didn't handle tuple keys it was already required that they be encoded as strings, right? We could fix that, too, and in my numpy-removing routine turn tuples into strings?

**Comment by bernstei** [comment]
In the julia dict format, e.g. `Dict((C, C) => (1.2, 4))`, is it really supposed to be that, or actually `Dict(("C", "C") => (1.2, 4))` ?


If the latter (and julia accepts strings in single quotes) I can just do `if isinstance(item, tuple): return str(tuple)` (which would return a string containing `('C', 'C')`), but if the former, it'd have to be something like `return "(" + ", ".join([str(subitem) for subitem in item]) + ")"` so it returns `(C, C)`

**Comment by gelzinyte** [comment]
yeah, can merge this. 

You're right, it's actually `Dict(("C", "C") => (1.2, 4))` in Julia. I think the problem was with serialising nested quotes, but julia also doesn't accept singly-quoted strings. I think let's not worry about this now...

**Comment by bernstei** [comment]
It was easy enough to implement.  I guess I'll add a test.

**Comment by bernstei** [comment]
OK, I got the tests fixed, and the right format for exporting tuples so that julia will read them correctly. I'm ready to merge, unless you have any other concerns.

**Pull request merged**